Rajitha/LoopsTest.js

```diff
@@ -0,0 +1,203 @@
+//};
+console.log(combineArray(arr,arr1));
+
+const alternateArray = (arr,arr1) => {
+    let altArray = [];
+    for (let i=0; i<arr.length; i++) {
+        altArray = altArray.concat(arr[i], arr1[i]);
+    }
+    return altArray;
+}
+console.log(alternateArray(arr,arr1));
+
+/**************************************************************
+ * write a function that returns all even number - [2,8,10,6]
+ * write a function that returns all odd numbers - [1,5]
+ * **************************************************************/
+const arr2 = [2,8,1,10,5,6];
+
+const evenOddArrays = arr => {
+    let evenArray = [];
+    let oddArray = [];
+    for (let i=0; i<arr.length; i++) {
+        if (arr[i]%2 == 0) {
+            evenArray.push(arr[i]);
+        } else {
+            oddArray.push(arr[i]);
+        }
+    }
+    console.log(evenArray);
+    console.log(oddArray);
+    return [evenArray, oddArray];
+};
+
+console.log(evenOddArrays(arr2)); //is there a way to return two arrays in a single function? multi-dimensinal array?
+
+/**************************************************************
+ * Using DO loop
+ * write a function that returns all even number - [2,8,10,6]
+ * write a function that returns all odd numbers - [1,5]
+ * **************************************************************/
+const evenArrayDoLoop = arr => {
+    let evenArray = [];
+    let i = 0;
+    do {
+        if (arr[i]%2 == 0) {
+            evenArray.push(arr[i]);
+        }
+        i++;
+    }
+    while (i<=arr.length);
+    return evenArray;
+}
+console.log(evenArrayDoLoop(arr2));
+
+/******* Do Loop - Odd Array ********/
+ const oddArrayDoLoop = arr => {
+    let oddArray = [];
+    let i = 0;
+    do {
+        if (arr[i]%2 != 0) {
+            oddArray.push(arr[i]);
+        }
+        i++;
+    }
+    while (i<=arr.length); //Why does it return undefined here but not in evenDoLoop????
+    return oddArray;
+}
+ console.log(oddArrayDoLoop(arr2));
+
+var arr3 = [
+    {type: "socks", quantity: 1},
+    {type: "shoes", quantity: 3},
+    {type: "pants", quantity: 1},
+    {type: "socks", quantity: 5},
+    {type: "shoes", quantity: 1},
+    {type: "socks", quantity: 10}
+];
+
+/******* Write a function that returns the total quantity of the socks ****/
+const totalSocks = arr => {
+    let socks = 0;
+    for (let i=0; i<arr.length; i++){
+        socks += arr[i].quantity;
+    }
+    return socks;
+}
+console.log("Total socks " +totalSocks(arr3));
+
+/***** Using forEach *****/
+let totSocks = 0;
+
+arr3.forEach((addSocks) => {
+    totSocks += addSocks.quantity;
+    console.log(totSocks);
+}); //Can we return the final count of socks???
+// let i=0 or i = 0 ????
+
+var str = "supercalifragilisticexpialidocious";
+
+/*** Write a function that returns the number of "a" in the str ***/
+const retACount = str => {
+    let aCount=0;
+    for (let i=0; i<str.length; i++){
+        if(str.charAt(i)==='a'){
```

Comment:
This is not wrong but want to bring up that since you are just checking if the value is the same and not trying to set a new letter for the value, it would be better performance to just use the index instead of calling the `.charAt` method.

Example:
https://stackoverflow.com/questions/5943726/string-charatx-or-stringx

Reply:
How woud you refactor this code to check for any single character? - Ren

Reply:
if(str[i].indexOf(ch) != -1){
            anyCount += 1;
        }

Refactor code to check for any single character
/*** Refactor to check any single character ***/
const retAnyCharCount = (str,ch) => {
    let anyCount=0;
    for (let i=0; i<str.length; i++){
        if(str[i].indexOf(ch) != -1){
            anyCount += 1;
        }
    }
    return anyCount;
}
console.log("Any Char Count in For loop is " + retAnyCharCount(str, 'i'));